src/components/Loading/Loading.react.js

```diff
@@ -0,0 +1,165 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import GraphSpinner from './spinners/GraphSpinner.jsx';
+import DefaultSpinner from './spinners/DefaultSpinner.jsx';
+import CubeSpinner from './spinners/CubeSpinner.jsx';
+import CircleSpinner from './spinners/CircleSpinner.jsx';
+import DotSpinner from './spinners/DotSpinner.jsx';
+import R from 'ramda';
+
+function getSpinner(type) {
+    switch (type) {
+        case 'graph':
+            return GraphSpinner;
+        case 'cube':
+            return CubeSpinner;
+        case 'circle':
+            return CircleSpinner;
+        case 'dot':
+            return DotSpinner;
+        default:
+            return DefaultSpinner;
+    }
+}
+
+/**
+ * A Loading component that wraps any other component and displays a spinner until the wrapped component has rendered.
+ */
+export default class Loading extends Component {
+    render() {
+        const {
+            loading_state,
+            color,
+            className,
+            style,
+            fullscreen,
+            debug,
+            type,
+        } = this.props;
+
+        const initial_loading_state = R.isNil(loading_state)
+            ? false
+            : loading_state.is_loading;
```

Comment:
The first impression I got from reading this variable's name was that this had to do with initial vs. subsequent loading. A more precise name would clarify the meaning.

Reply:
In a way it is, because the `initial_loading_state` is the loading state of the Loading component itself. Any suggestions? Not sure what naming would be more precise.

Reply:
Would probably just make it `isLoading` like elsewhere